src/infra/core/src/ingest/fetch_service/evm.rs

```diff
@@ -0,0 +1,300 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::HashMap;
+use std::path::Path;
+use std::sync::Arc;
+
+use futures::TryStreamExt;
+use internal_error::{ErrorIntoInternal, ResultIntoInternal};
+use kamu_core::*;
+use kamu_datasets::DatasetEnvVar;
+use opendatafabric::*;
+
+use super::*;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+impl FetchService {
```

Comment:
Note: Interesting approach, with the implementation of different methods, in different modules

Reply:
I would not recommend this pattern. This is just a transitional step towards separating these into different `Source` types.

I think multiple `impl` blocks are justified only in `opendatafabric` where we add extra functionality to code-generated types.